### ISSUE TITLE
[Runtime Error] Assertion failed

### ISSUE THREAD

**Issue by andyrichardson** [koush/vysor.io#339]
### About

The following runtime error is presented when multiple displays are connected via a DisplayLink dock.

```
Assertion failed!

Program: ...p-1.9.3.crx-unpacked\native\win32\video_decode.dll
File: video_decode.cc
Line: 704

Expression: !context_->is_null()
```

<details>
  <summary>Click for screenshot</summary>
  <img src="https://user-images.githubusercontent.com/10779424/43449356-218d131c-94a8-11e8-90a4-08b9184c94a5.PNG" />
</details>

### Reproduction

1. Download and run the Windows binary
2. Connect phone via USB
3. Click view (under "Choose a device")

### Expected result

Android screen is mirrored.

### Actual result

Black screen with error.

### Additional information

 * Windows 10
 * Dell D3100 USB3.0 Dock
 * Display Link Driver v8.5.3459.0
 * This "Display Link" issue isn't present on any other software

**Comment by andyrichardson** [comment]
#286

**Comment by josiasfabricio** [comment]
Try run Windows 8 Compatibility Mode

**Comment by andyrichardson** [comment]
Thanks @josiasfabricio. 

I'm away from the office now but I'll try that out some time next week and provide an update.

**Comment by andyrichardson** [comment]
Unfortunately Windows 8 Compatibility Mode didn't fix the issue 👎

**Comment by slewis69au** [comment]
I have exactly the same issue - no resolution using any of the suggested solutions.  Running a Dell i7 laptop with a usbc dock.  High performance and compatibility modes have no impact.

**Comment by NegativeIQ** [comment]
Same problem here on Windows 7 x64 SP1 with Visual Studio installed and Zuk Z1 CyanogenMod. By the way no dock or anything just USB cable directly to laptop (3,0/3.1 port).

**Comment by gonzobrains** [comment]
Is Vysor still supported?  I have reported this issue months ago but have never received a response despite having a paid pro version of the app.

**Comment by koush** [comment]
Multiple monitors by chance?

**Comment by gonzobrains** [comment]
Yes, for the most part.  It's hit or miss.  Sometimes windows 7
compatibility works, sometimes running as administrator works, but not
always.

What can I do?  I really want to go back to my pro license.  Any ideas?

**Comment by koush** [comment]
I'm going to see if I can remove the WebGL requirement.

**Comment by CaptainXLAB** [comment]
I can confirm multiple monitors cause this too. I have a desktop, single GPU, the same error on vysor desktop app (had it before in chrome too but not anymore for whatever reason).

**Run it by physically plugging the other monitor out**, click "view" and it starts without problems, **plug the monitor back in and it works fine after that.**

**Comment by c0mmoncents** [comment]
I'm having the same issue since signing up for the pro version. Single monitor i7 4770 16gb ram geforce gtx 1660 6 gb gaming oc.

**Comment by gonzobrains** [comment]
Any updates on this?  Also, is there a way to transfer a license from one
Google Account to another?  I bought with my personal account but I end up
using this on my work computer.

**Comment by CaptainXLAB** [comment]
For Vysor Chrome - Launch Chrome first. Open google or any random website. Always keep the main window open. Then launch vysor from chrome commandline, while keeping the main chrome window open:

"C:\Program Files (x86)\Google\Chrome\Application\chrome_proxy.exe" --profile-directory=Default --app-id=gidgenkbbabolejbgbpnhbimgjbffefm

If I launch Chrome first, it doesn't give that error, works flawlessly. But if all chrome tabs are closed, it gives that error again.

**Comment by koush** [comment]
Can you try the new progressive web app beta here:
https://app.vysor.io
I want to verify this is fixed.

Native apps coming soon.

**Comment by c0mmoncents** [comment]
web app works fine for me

**Comment by koush** [comment]
> web app works fine for me

Mirrors well? No black screen? How is high quality performance?

**Comment by c0mmoncents** [comment]
Works well. No black screen. Quailty seemed great opened games amd everything looked good.

**Comment by koush** [comment]
@c0mmoncents awesome thanks. in case you didn't notice, that pwa is installable to the desktop and chrome launcher via the 3 dots menu, and then Install Vysor.

There will be a native app (which will run a bit better, and have a few more features due to less limitations) sometime soon as well.

**Comment by ridimdev** [comment]
> Try run Windows 8 Compatibility Mode

It worked for me!

**Comment by gonzobrains** [comment]
What would be a permanent Windows 10 solution?

**Comment by brainv** [comment]
Same here, still not working in windows 10 :(

**Comment by huangliqiang** [comment]
Reproduction:
1.work in Vmware. windows 10 or windows 7 
2.Vmware workstation/ Vmware Esxi6.7.2
3.USB3.0 / USB 2.0
![44771965-dd27f080-ab6c-11e8-9057-edd7d2678d8c](https://user-images.githubusercontent.com/6927326/100213180-49128280-2f49-11eb-97db-7c6b405d2cfd.png)

Solution:
1.安装Vmware tools
2.编辑设置虚拟机--开启3D支持
![photo_2020-11-26_02-09-57](https://user-images.githubusercontent.com/6927326/100213390-85de7980-2f49-11eb-8c81-b06851a305ec.jpg)

**Comment by atuldpatil** [comment]
In my case when I launched installed application (Vysor.exe) it was giving 
"Assertion failed! in video_decode.dll in File video_decode.cc at line 327 with Expression: result == PP_OK"

When I tried adding Chrome extension [Vysor](https://chrome.google.com/webstore/detail/vysor/gidgenkbbabolejbgbpnhbimgjbffefm)
it gave error: 

> "WebGL is not supported"

However I was seeing Vysor in `chrome://apps/` and when I launched that and clicked "VIEW" device, I got error:

> "Error connecting to Android device. Vysor was unable to access the Android USB device. Vysor may be mirroring this Android in another tab"

Then I killed all running instances of "Vysor.exe" and an instance of "adb.exe"  from task manager. (Make sure to kill in that order. Because if you kill adb.exe first, then running instance of Vysor.exe might start it again)

Then in Chrome I made these settings

Under: [chrome://settings/](chrome://settings/)
Use hardware acceleration when available: True


Under [chrome://flags/](chrome://flags/) 
WebGL 2.0 Compute	Enabled
WebGL Draft Extensions	Enabled
Override software rendering list Enabled
Hardware-accelerated video decode (Already enabled)

Restart Chrome.

And now when I launched from here
[chrome://apps/](chrome://apps/)
It works like charm 
(However I am still getting assertion error if I try to launch Visor installed application (Vysor.exe). But since its working via Chrome extension, I don't care about it now)
